test/test-find-node-directory.js

```diff
@@ -0,0 +1,98 @@
+var test = require('tape')
+var path = require('path')
+var findNodeDirectory = require('../lib/find-node-directory')
+
+// we should find the directory based on the directory
+// the script is running in and it should match the layout
+// in a build tree where npm is installed in
+// .... /deps/npm
+test('test find-node-directory - node install', function (t) {
+  t.plan(1)
+  var processObj = {execPath: '/x/y/bin/node', platform: process.platform}
+  t.equal(
+    findNodeDirectory('/x/deps/npm/node_modules/node-gyp/lib', processObj),
+                      path.join('/x'));
+})
+
+// we should find the directory based on the directory
+// the script is running in and it should match the layout
+// in a installed tree where npm is installed in
+// .... /lib/node_modules/npm or .../node_modules/npm
+// dependingo on the patform
+test('test find-node-directory - node build', function (t) {
+  t.plan(1)
+  var processObj = {execPath: '/x/y/bin/node', platform: process.platform}
+  if (process.platform === 'win32') {
+    t.equal(
+      findNodeDirectory('/y/node_modules/npm/node_modules/node-gyp/lib',
+                         processObj), path.join('/y'));
+  } else {
+    t.equal(
+      findNodeDirectory('/y/lib/node_modules/npm/node_modules/node-gyp/lib',
+                         processObj), path.join('/y'));
+  }
```

Comment:
I'd recommend doing that everywhere where you call findNodeDirectory() with `{ platform: process.platform }`.  The idea is that all code paths are tested on all platforms.